components/hub75_matrix_display/matrix_display.cpp

```diff
@@ -76,46 +87,139 @@ namespace esphome
             case HUB75_I2S_CFG::shift_driver::MBI5124:
                 ESP_LOGCONFIG(TAG, "  Driver: MBI5124");
                 break;
-            case HUB75_I2S_CFG::shift_driver::SM5266P:
-                ESP_LOGCONFIG(TAG, "  Driver: SM5266P");
-                break;
-            case HUB75_I2S_CFG::shift_driver::DP3246_SM5368:
-                ESP_LOGCONFIG(TAG, "  Driver: DP3246_SM5368");
+            case HUB75_I2S_CFG::shift_driver::DP3246:
+                ESP_LOGCONFIG(TAG, "  Driver: DP3246");
                 break;
             }
 
             ESP_LOGCONFIG(TAG, "  I2S Speed: %u MHz", (uint32_t)cfg.i2sspeed / 1000000);
             ESP_LOGCONFIG(TAG, "  Latch Blanking: %i", cfg.latch_blanking);
             ESP_LOGCONFIG(TAG, "  Clock Phase: %s", TRUEFALSE(cfg.clkphase));
             ESP_LOGCONFIG(TAG, "  Min Refresh Rate: %i", cfg.min_refresh_rate);
+
+#ifdef USE_VIRTUAL_PANEL
+            ESP_LOGCONFIG(TAG, "  VirtualMatrix: %s", this->use_virtual_panel_ ? "ENABLED" : "disabled");
+            if (this->use_virtual_panel_) {
+                ESP_LOGCONFIG(TAG, "    Grid: %ux%u panels (total %u)",
+                              (unsigned)this->v_cols_, (unsigned)this->v_rows_,
+                              (unsigned)(this->v_cols_ * this->v_rows_));
+            }
+#endif
         }
 
         void MatrixDisplay::set_brightness(int brightness)
         {
-            // Wrap brightness function
-            this->dma_display_->setBrightness8(brightness);
+            // Remember last non-zero so power-on can restore without fighting the number entity
+            if (brightness > 0) this->stored_brightness_ = brightness;
+            // Only apply to hardware when enabled; if off, just remember it
+            if (this->enabled_ && this->dma_display_ != nullptr) {
+                this->dma_display_->setBrightness8(brightness);
+            }
         }
 
         void HOT MatrixDisplay::draw_absolute_pixel_internal(int x, int y, Color color)
         {
+            if (!this->dma_display_) return;
+            if (!this->enabled_) return;
```

Comment:
I think we should allow changes to the buffer even if the display is off. This would ensure that the display does not show "outdated information" on the first "frame" after being enabled.

The "enabled-guard" pops up in a few more places.